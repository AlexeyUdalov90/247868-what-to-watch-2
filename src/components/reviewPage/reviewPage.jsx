import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';

import Header from '../header/header.jsx';
import {Operation} from '../../redusers/reducer.js';
import getFilm from '../../selectors/getFilm.js';

import withReviewForm from '../../hocs/withReviewForm/withReviewForm.js';

const ReviewPage = ({film, isAuthorizationRequired, avatarUrl, disabledButton, comment, onChangeCommentHandler, onChangeRatingHandler, onSubmitHandler}) => {
  const {backgroundImage, name, posterImage, id} = film;

  return (<section className="movie-card movie-card--full">
    <div className="movie-card__header">
      <div className="movie-card__bg">
        <img src={backgroundImage} alt={name} />
      </div>

      <h1 className="visually-hidden">WTW</h1>

      <Header isAuthorizationRequired={isAuthorizationRequired} avatarUrl={avatarUrl} personalClass={``}>
        <nav className="breadcrumbs">
          <ul className="breadcrumbs__list">
            <li className="breadcrumbs__item">
              <Link to={`/films/${id}`} className="breadcrumbs__link">{name}</Link>
            </li>
            <li className="breadcrumbs__item">
              <a className="breadcrumbs__link">Add review</a>
            </li>
          </ul>
        </nav>
      </Header>

      <div className="movie-card__poster movie-card__poster--small">
        <img src={posterImage} alt={name} width="218" height="327" />
      </div>
    </div>

    <div className="add-review">
      <form action="#" className="add-review__form" onSubmit={onSubmitHandler}>
        <div className="rating">
          <div className="rating__stars">
            <input className="rating__input" id="star-1" type="radio" name="rating" value="1" />
            <label className="rating__label" forhtml="star-1" onClick={onChangeRatingHandler}>Rating 1</label>

            <input className="rating__input" id="star-2" type="radio" name="rating" value="2" />
            <label className="rating__label" forhtml="star-2" onClick={onChangeRatingHandler}>Rating 2</label>

            <input className="rating__input" id="star-3" type="radio" name="rating" value="3" />
            <label className="rating__label" forhtml="star-3" onClick={onChangeRatingHandler}>Rating 3</label>

            <input className="rating__input" id="star-4" type="radio" name="rating" value="4" />
            <label className="rating__label" forhtml="star-4" onClick={onChangeRatingHandler}>Rating 4</label>

            <input className="rating__input" id="star-5" type="radio" name="rating" value="5" />
            <label className="rating__label" forhtml="star-5" onClick={onChangeRatingHandler}>Rating 5</label>
          </div>
        </div>

        <div className="add-review__text">
          <textarea className="add-review__textarea" name="review-text" id="review-text" placeholder="Review text" minLength="50" maxLength="400" value={comment} onChange={onChangeCommentHandler}></textarea>
          <div className="add-review__submit">
            <button className="add-review__btn" type="submit" disabled={disabledButton ? `disabled` : ``}>Post</button>
          </div>
        </div>
      </form>
    </div>
  </section>);
};

ReviewPage.propTypes = {
  film: PropTypes.object,
  isAuthorizationRequired: PropTypes.bool,
  avatarUrl: PropTypes.string,
  sendReview: PropTypes.func,
  disabledButton: PropTypes.bool,
  rating: PropTypes.number,
  comment: PropTypes.string,
  onChangeCommentHandler: PropTypes.func,
  onChangeRatingHandler: PropTypes.func,
  onSubmitHandler: PropTypes.func,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, state, {
  film: getFilm(state, ownProps),
  isAuthorizationRequired: state.load.isAuthorizationRequired,
  avatarUrl: state.load.userData.avatarUrl,
});

const mapDispatchToProps = (dispatch) => {
  return {
    sendReview: (id, rating, comment) => {
      dispatch(Operation.sendReview(id, rating, comment));
    },
  };
};

export {ReviewPage};

export default connect(mapStateToProps, mapDispatchToProps)(withReviewForm(ReviewPage));
