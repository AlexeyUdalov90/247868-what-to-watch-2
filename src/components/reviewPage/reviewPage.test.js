import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import {ReviewPage} from './reviewPage.jsx';

it(`ReviewPage correctly renders after relaunch`, () => {
  const tree = renderer.create(<Router><ReviewPage
    film={{}}
    isAuthorizationRequired={false}
    avatarUrl={``}
    disabledButton={false}
    comment={``}
    onChangeCommentHandler={jest.fn()}
    onChangeRatingHandler={jest.fn()}
    onSubmitHandler={jest.fn()}
  /></Router>).toJSON();

  expect(tree).toMatchSnapshot();
});
