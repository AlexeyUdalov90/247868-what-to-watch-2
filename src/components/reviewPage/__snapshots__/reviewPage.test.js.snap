// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewPage correctly renders after relaunch 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header "
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/undefined"
              onClick={[Function]}
            />
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <a
            href="/mylist"
            onClick={[Function]}
          >
            <img
              alt="User avatar"
              height="63"
              src=""
              width="63"
            />
          </a>
        </div>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        height="327"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[MockFunction]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            type="radio"
            value="1"
          />
          <label
            className="rating__label"
            forhtml="star-1"
            onClick={[MockFunction]}
          >
            Rating 1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            type="radio"
            value="2"
          />
          <label
            className="rating__label"
            forhtml="star-2"
            onClick={[MockFunction]}
          >
            Rating 2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            type="radio"
            value="3"
          />
          <label
            className="rating__label"
            forhtml="star-3"
            onClick={[MockFunction]}
          >
            Rating 3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            type="radio"
            value="4"
          />
          <label
            className="rating__label"
            forhtml="star-4"
            onClick={[MockFunction]}
          >
            Rating 4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            type="radio"
            value="5"
          />
          <label
            className="rating__label"
            forhtml="star-5"
            onClick={[MockFunction]}
          >
            Rating 5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          maxLength="400"
          minLength="50"
          name="review-text"
          onChange={[MockFunction]}
          placeholder="Review text"
          value=""
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled=""
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
